### PR TITLE
[Tables] Add SAS to tables

### PR DESCRIPTION
**Pull request by seankane-msft**
#16610

### CODE REVIEW

File: sdk/tables/azure-data-tables/azure/data/tables/_models.py [hunk 1]
```diff
@@ -611,11 +611,12 @@ class ResourceTypes(object):
     """
 
     def __init__(
-        self, service=False, object=False
+        self, service=False, object=False, container=False
```
Comment (annatisch) [inline sdk/tables/azure-data-tables/azure/data/tables/_models.py:614]: ummm why did we add container? Containers aren't a resource for Tables :)

File: sdk/tables/azure-data-tables/azure/data/tables/aio/_base_client_async.py [hunk 2]
```diff
@@ -27,6 +28,7 @@
     HttpLoggingPolicy,
     UserAgentPolicy,
     ProxyPolicy,
+    AzureSasCredentialPolicy
```
Comment (annatisch) [inline sdk/tables/azure-data-tables/azure/data/tables/aio/_base_client_async.py:31]: These are the same types for sync/async right?
Pretty sure we had no async-specific types for this auth...

File: sdk/tables/azure-data-tables/tests/test_table_entity.py [hunk 3]
```diff
@@ -1860,12 +1879,19 @@ def test_sas_signed_identifier(self, tables_storage_account_name, tables_primary
...
             identifiers = {'testid': access_policy}
 
             self.table.set_table_access_policy(identifiers)
 
+            # token = self.get_sas(
+            #     tables_storage_account_name,
+            #     tables_primary_storage_account_key,
+            #     policy_id='testid'
+            # )
+            # self.scrubber.register_name_pair(token, "fake_token_value")
```
Comment (annatisch) [inline sdk/tables/azure-data-tables/tests/test_table_entity.py:1893]: This wasn't working?

File: sdk/tables/azure-data-tables/tests/test_table_entity_async.py [hunk 4]
```diff
@@ -1829,7 +1829,7 @@ async def test_sas_upper_case_table_name(self, tables_storage_account_name, tabl
         finally:
             await self._tear_down()
 
-    @pytest.mark.skip("Header authorization is malformed")
+    # @pytest.mark.skip("Header authorization is malformed")
```
Comment (annatisch) [inline sdk/tables/azure-data-tables/tests/test_table_entity_async.py:1832]: Can can now remove this?

File: sdk/tables/azure-data-tables/azure/data/tables/aio/_base_client_async.py (hunk 2 above)
Comment (seankane-msft) [reply sdk/tables/azure-data-tables/azure/data/tables/aio/_base_client_async.py:31]: Yes you are correct

File: sdk/tables/azure-data-tables/tests/test_table_entity.py (hunk 3 above)
Comment (seankane-msft) [reply sdk/tables/azure-data-tables/tests/test_table_entity.py]: No, this is a separate issue where SAS tokens are stored in plain text in recording files. Opening a separate issue/PR for this

**Review by annatisch — APPROVED** [review]
LGTM - do you want to add a note to the changelog?
Or do that separately?

**Pull request merged**